### PR TITLE
fix(executor): The logic to deal with tx_hook and tx_body

### PR DESCRIPTION
**Pull request by LycrusHamster**
fix: fix the logic to deal with tx_hook and tx_body

**What type of PR is this?**
fix: fix executor, change the logic/sequence of dealing with tx hooks and tx body

**What this PR does / why we need it**:
this logic should fit normal business better

### CODE REVIEW

File: framework/src/executor/mod.rs [hunk 1]
```diff
@@ -397,41 +398,47 @@ impl<S: 'static + Storage, DB: 'static + TrieDB, Mapping: 'static + ServiceMappi
...
-
         if ret.is_error() {
             service_context.cancel("tx_exec_return_code_not_zero".to_owned());
         }
 
-        tx_hooks.after(context, service_context)?;
-        self.states.stash()?;
+        let resp = tx_hooks.after(context, service_context)?;
+
+        if resp.iter().filter(|r| r.is_error()).count() > 0 {
```
Comment (KaoImin) [inline framework/src/executor/mod.rs:436]:
```suggestion
        if resp.iter().any(|r| r.is_error()) {
```

**Comment by yejiayu** [comment]
/cc @KaoImin 

Does this PR conflict with https://github.com/nervosnetwork/muta/pull/365?
If true, we should plan sort of merging.

File: framework/src/executor/mod.rs (hunk 1 above)
Comment (LycrusHamster) [reply framework/src/executor/mod.rs]: good idea

**Comment by KaoImin** [comment]
> /cc @KaoImin @LycrusHamster
> 
> Does this PR conflict with #365?
> If true, we should plan a sort of merging.

This PR should be merged after #365

File: framework/src/executor/tests/framework.rs [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+use protocol::traits::{
+    Context, Executor, ExecutorParams, ExecutorResp, Service, ServiceMapping, ServiceResponse,
+    ServiceSDK, Storage,
+};
+use protocol::types::{
+    Address, Block, Genesis, Hash, Proof, RawTransaction, Receipt, ServiceContext,
+    SignedTransaction, TransactionRequest,
+};
+use protocol::ProtocolResult;
+use std::sync::Arc;
```
Comment (yejiayu) [inline framework/src/executor/tests/framework.rs:15]:
```suggestion
use std::sync::Arc;

use async_trait::async_trait;
use bytes::{Bytes, BytesMut};
use cita_trie::MemoryDB;

use super::super::super::executor::ServiceExecutor;

use binding_macro::{cycles, service, tx_hook_after, tx_hook_before};
use protocol::traits::{
    Context, Executor, ExecutorParams, ExecutorResp, Service, ServiceMapping, ServiceResponse,
    ServiceSDK, Storage,
};
use protocol::types::{
    Address, Block, Genesis, Hash, Proof, RawTransaction, Receipt, ServiceContext,
    SignedTransaction, TransactionRequest,
};
use protocol::ProtocolResult;
```

and the `super` keyword is disabled

Comment (LycrusHamster) [reply framework/src/executor/tests/framework.rs]: solved

**Pull request merged**